base/ca/src/main/java/com/netscape/ca/CertificateAuthority.java

```diff
@@ -1037,19 +1036,25 @@ public X509CertImpl getCACert() throws EBaseException {
             return caCertImpl;
         }
 
-        String cert = mConfig.getString("signing.cert");
-        logger.debug("CertificateAuthority: CA signing cert: " + cert);
+        String certnickname = mConfig.getString("signing.certnickname");
+        String tokennickname = mConfig.getString("signing.tokenname");
 
-        if (StringUtils.isEmpty(cert)) {
+        String certName = tokennickname + ":" + certnickname;
+        logger.debug("CertificateAuthority: Getting CA signing cert: " + certName);
+
+        CryptoManager manager;
+        X509Certificate caCert;
+        try {
+            manager= CryptoManager.getInstance();
+            caCert = manager.findCertByNickname(certName);
+        } catch (ObjectNotFoundException | NotInitializedException | TokenException e) {
             logger.error("CertificateAuthority: Missing CA signing certificate");
-            throw new EBaseException("Missing CA signing certificate");
+            throw new EBaseException("Missing CA signing certificate", e);
```

Comment:
As indicated by the list of exceptions to catch, there could be several reasons why the code fails, so it might be better to generate a more generic logging message like this:
```
logger.error("CertificateAuthority: Unable to find CA signing certificate: " + e.getMessage(), e);
throw new EBaseException("Unable to find CA signing certificate: " + e.getMessage(), e);
```